### PR TITLE
VOTE-2810, VOTE-2811, VOTE-2812 state content field changes

### PR DESCRIPTION
**Pull request by daniel-crowder**
## Jira ticket

[VOTE-2810](https://cm-jira.usa.gov/browse/VOTE-2810) - New field type
[VOTE-2811](https://cm-jira.usa.gov/browse/VOTE-2811) - New block type
[VOTE-2812](https://cm-jira.usa.gov/browse/VOTE-2812) - Update State Content type

## Description

Add a new custom compound field. Add new fields to the State content type using the new compound field. Create a new block type using the new compound field along with other field types.

### Post-deploy steps

1. Run 'lando retune' locally

### QA/Testing instructions

1. Login as an admin
2. Goto content admin page and edit a state content item
3. Locate the new tab, "State display content"
4. Confirm the available fields match the document located in https://cm-jira.usa.gov/browse/VOTE-2811
5. Add content to the fields and confirm that it saves
6. Edit the state again and confirm that the fields display in the fields
7. Goto content admin page again and go to block types
8. Locate the "State Display Content" link
9. Click the manage fields link and confirm the available fields match the document located in https://cm-jira.usa.gov/browse/VOTE-2812
10. Go back to the block types page and click the button up top to create a new block
11. Add a "State Display Content" block type
12. Fill out the form, save, and confirm that the block saves successfully
13. Edit the block and confirm that the fields display

## Checklist for the Developer

- [x] A link to the JIRA ticket has been included above.
- [x] No merge conflicts exist with the target branch.
- [x] Automated tests have passed on this PR.
- [x] A reviewer has been designated.
- [x] Deployment and testing steps have been documented above, if applicable.

## Checklist for the Peer Reviewers

- [ ] The file changes are relevant to the task objective.
- [ ] Code is readable and includes appropriate commenting.
- [ ] Code standards and best practices are followed.
- [ ] QA/Test steps were successfully completed, if applicable.
- [ ] Applicable logs are free of errors.

### CODE REVIEW

**Review by rayestrada — CHANGES_REQUESTED** [review]

File: config/sync/field.field.node.state_territory.field_registration_not_needed.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  config:
+    - field.storage.node.field_registration_not_needed
+    - node.type.state_territory
+  module:
+    - vote_fields
+id: node.state_territory.field_registration_not_needed
+field_name: field_registration_not_needed
+entity_type: node
+bundle: state_territory
+label: 'Registration not needed'
```
[the remark this answers is not in the document]
Comment (daniel-crowder) [reply config/sync/field.field.node.state_territory.field_registration_not_needed.yml:14]: Updated!

File: config/sync/field.field.node.state_territory.field_military_and_overseas_regi.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  config:
+    - field.storage.node.field_military_and_overseas_regi
+    - node.type.state_territory
+  module:
+    - vote_fields
+id: node.state_territory.field_military_and_overseas_regi
+field_name: field_military_and_overseas_regi
+entity_type: node
+bundle: state_territory
+label: 'Military and overseas registration'
```
[the remark this answers is not in the document]
Comment (daniel-crowder) [reply config/sync/field.field.node.state_territory.field_military_and_overseas_regi.yml]: Updated!

File: config/sync/field.field.node.state_territory.field_in_person_registration.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  config:
+    - field.storage.node.field_in_person_registration
+    - node.type.state_territory
+  module:
+    - vote_fields
+id: node.state_territory.field_in_person_registration
+field_name: field_in_person_registration
+entity_type: node
+bundle: state_territory
+label: 'In person registration'
```
[the remark this answers is not in the document]
Comment (daniel-crowder) [reply config/sync/field.field.node.state_territory.field_in_person_registration.yml]: Updated!

File: config/sync/field.field.node.state_territory.field_mail_registration.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  config:
+    - field.storage.node.field_mail_registration
+    - node.type.state_territory
+  module:
+    - vote_fields
+id: node.state_territory.field_mail_registration
+field_name: field_mail_registration
+entity_type: node
+bundle: state_territory
+label: 'Mail registration'
```
[the remark this answers is not in the document]
Comment (daniel-crowder) [reply config/sync/field.field.node.state_territory.field_mail_registration.yml]: Updated!

File: config/sync/field.field.node.state_territory.field_online_registration.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  config:
+    - field.storage.node.field_online_registration
+    - node.type.state_territory
+  module:
+    - vote_fields
+id: node.state_territory.field_online_registration
+field_name: field_online_registration
+entity_type: node
+bundle: state_territory
+label: 'Online registration'
```
[the remark this answers is not in the document]
Comment (daniel-crowder) [reply config/sync/field.field.node.state_territory.field_online_registration.yml]: Updated!

File: config/sync/field.field.node.state_territory.field_check_registration.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  config:
+    - field.storage.node.field_check_registration
+    - node.type.state_territory
+  module:
+    - vote_fields
+id: node.state_territory.field_check_registration
+field_name: field_check_registration
+entity_type: node
+bundle: state_territory
+label: 'Check registration'
```
[the remark this answers is not in the document]
Comment (daniel-crowder) [reply config/sync/field.field.node.state_territory.field_check_registration.yml]: Updated!

File: web/modules/custom/vote_fields/src/Plugin/Field/FieldWidget/StateContentWidget.php [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      '#allowed_formats' => ['simple_html'],
+    ];
+
+    $element['link_text'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Link text'),
+      '#default_value' => $items[$delta]->link_text ?? NULL,
+    ];
+
+    $element['#theme_wrappers'] = ['container', 'form_element'];
```
[the remark this answers is not in the document]
Comment (daniel-crowder) [reply web/modules/custom/vote_fields/src/Plugin/Field/FieldWidget/StateContentWidget.php]: Updated!

**Review by rayestrada — APPROVED** [review]

**Pull request merged**